From 53516f420745a1c8b5ddd081c61c919336fe2ef7 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Noralf=20Tr=C3=B8nnes?= <noralf@tronnes.org>
Date: Thu, 28 May 2020 15:57:57 +0200
Subject: hack: Option to force RG16

---
 drivers/gpu/drm/gud/gud_gadget.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/gpu/drm/gud/gud_gadget.c b/drivers/gpu/drm/gud/gud_gadget.c
index e5e7f56889fc..fddab807a643 100644
--- a/drivers/gpu/drm/gud/gud_gadget.c
+++ b/drivers/gpu/drm/gud/gud_gadget.c
@@ -22,6 +22,9 @@
 
 #include "gud_internal.h"
 
+static bool gud_gadget_force_rg16;
+module_param_named(force_rg16, gud_gadget_force_rg16, bool, 0600);
+
 /*
  * Concurrency:
  * Calls into this module from f_gud_drm are serialized and run in process
@@ -785,6 +788,9 @@ static int gud_drm_gadget_get_formats(struct gud_drm_gadget *gdg, u8 *max_cpp)
 		if (info->num_planes != 1)
 			continue;
 
+		if (gud_gadget_force_rg16 && info->format != DRM_FORMAT_RGB565)
+			continue;
+
 		format = gud_from_fourcc(info->format);
 		if (!format)
 			continue;
-- 
2.23.0

